src/app/core/admin/admin-entity-details/admin-entity-details/admin-entity-details.component.ts

```diff
@@ -44,15 +46,38 @@ export class AdminEntityDetailsComponent {
   @Input() entityConstructor: EntityConstructor;
   @Input() config: EntityDetailsConfig;
 
+  constructor(private dialog: MatDialog) {}
+
   newPanelFactory(): Panel {
     return { title: "New Tab", components: [] };
   }
 
   addComponent(panel: Panel) {
-    panel.components.push({
-      title: $localize`:Default title:New Section`,
-      component: "Form", // TODO: make this configurable
-      config: { fieldGroups: [] },
-    });
+    this.dialog
+      .open(EntityComponentSelectComponent, {
+        height: "20vh",
+      })
+      .afterClosed()
+      .subscribe((selectedSection) => {
+        if (!selectedSection) {
+          return;
+        }
+
+        if (selectedSection === "default-form") {
+          panel.components.push({
+            title: $localize`:Default title:New Section`,
+            component: "Form", // TODO: make this configurable
+            config: { fieldGroups: [] },
+          });
+        } else if (selectedSection === "related-form") {
+          panel.components.push({
+            title: $localize`:Default title:New Related Section`,
+            component: "RelatedEntities",
+            config: {
+              entityType: this.entityConstructor.ENTITY_TYPE, //we can not use empty string
+            },
+          });
+        }
```

Comment:
should we move this part into the dialog component, so that it returns a fully usable config object? If we get more and more different components, that may be more scalable